client-rpc/src/wallet_rpc.rs

```diff
@@ -0,0 +1,364 @@
+                    ExtendedAddr::BasicRedeem(address) => Ok(format!("{}", address)),
+                    _ => Err(rpc_error_from_string(
+                        "Unsupported address format".to_owned(),
+                    )),
+                })
+                .collect(),
+            Err(e) => Err(to_rpc_error(e)),
+        }
+    }
+
+    fn balance(&self, request: WalletRequest) -> jsonrpc_core::Result<Coin> {
+        self.sync()?;
+
+        match self.client.balance(&request.name, &request.passphrase) {
+            Ok(balance) => Ok(balance),
+            Err(e) => Err(to_rpc_error(e)),
+        }
+    }
+
+    fn create(&self, request: WalletRequest) -> jsonrpc_core::Result<String> {
+        if let Err(e) = self.client.new_wallet(&request.name, &request.passphrase) {
+            return Err(to_rpc_error(e));
+        }
+
+        if let Err(e) = self.client.new_address(&request.name, &request.passphrase) {
+            Err(to_rpc_error(e))
+        } else {
+            Ok(request.name.clone())
+        }
+    }
+
+    fn list(&self) -> jsonrpc_core::Result<Vec<String>> {
+        match self.client.wallets() {
+            Ok(wallets) => Ok(wallets),
+            Err(e) => Err(to_rpc_error(e)),
+        }
+    }
+
+    fn sendtoaddress(
+        &self,
+        request: WalletRequest,
+        to_address: String,
+        amount: u64,
+    ) -> jsonrpc_core::Result<()> {
+        self.sync()?;
+
+        let redeem_address = RedeemAddress::from_str(&to_address[..])
+            .map_err(|err| rpc_error_from_string(format!("{}", err)))?;
+        let address = ExtendedAddr::BasicRedeem(redeem_address);
+        let coin = Coin::new(amount).map_err(|err| rpc_error_from_string(format!("{}", err)))?;
+        let tx_out = TxOut::new(address, coin);
+        let tx_attributes = TxAttributes::new(self.chain_id);
+
+        if let Err(e) = self.client.create_and_broadcast_transaction(
+            &request.name,
+            &request.passphrase,
+            vec![tx_out],
+            tx_attributes,
+        ) {
+            Err(to_rpc_error(e))
+        } else {
+            Ok(())
+        }
+    }
+
+    fn sync(&self) -> jsonrpc_core::Result<()> {
+        if let Err(e) = self.client.sync() {
+            Err(to_rpc_error(e))
+        } else {
+            Ok(())
+        }
+    }
+
+    fn sync_all(&self) -> jsonrpc_core::Result<()> {
+        if let Err(e) = self.client.sync_all() {
+            Err(to_rpc_error(e))
+        } else {
+            Ok(())
+        }
+    }
+
+    fn transactions(&self, request: WalletRequest) -> jsonrpc_core::Result<Vec<TransactionChange>> {
+        self.sync()?;
+
+        match self.client.history(&request.name, &request.passphrase) {
+            Ok(transaction_change) => Ok(transaction_change),
+            Err(e) => Err(to_rpc_error(e)),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct WalletRequest {
+    name: String,
+    passphrase: String,
```

Comment:
Is the passphrase cleared?

It'd be better to use something like https://github.com/myfreeweb/secstr

Reply:
client-core takes `&str`, so I guess this would be a bigger task -- perhaps put a TODO/WARNING comment and open a task for @devashishdxt to convert passphrase's strs to use secstrs or equivalents

Reply:
I am looking into the `secstr`. It seems not very difficult to integrate but do require some work to change the underlying wallet client implementation. I will leave a TODO here first and open a new task for it.